src/core/skills/characters/yijiang2014/qiangzhi.ts

```diff
@@ -0,0 +1,133 @@
+    const ownerId = content.fromId;
+
+    room.notify(
+      GameEventIdentifiers.AskForChoosingPlayerEvent,
+      EventPacker.createUncancellableEvent<GameEventIdentifiers.AskForChoosingPlayerEvent>({
+        triggeredBySkills: [this.GeneralName],
+        toId: ownerId,
+        players: room.AlivePlayers.filter(
+          player => player.getCardIds(PlayerCardsArea.HandArea).length > 0 && player.Id !== ownerId,
+        ).map(player => player.Id),
+        requiredAmount: 1,
+        conversation: TranslationPack.translationJsonPatcher(
+          '{0}: please choose a player to show a hand card',
+          this.GeneralName,
+        ).extract(),
+      }),
+      ownerId,
+    );
+
+    const { selectedPlayers } = await room.onReceivingAsyncResponseFrom(
+      GameEventIdentifiers.AskForChoosingPlayerEvent,
+      ownerId,
+    );
+
+    if (selectedPlayers === undefined) {
+      return false;
+    }
+
+    room.notify(
+      GameEventIdentifiers.AskForChoosingCardFromPlayerEvent,
+      EventPacker.createUncancellableEvent<GameEventIdentifiers.AskForChoosingCardFromPlayerEvent>({
+        triggeredBySkills: [this.GeneralName],
+        fromId: ownerId,
+        toId: selectedPlayers[0],
+        options: {
+          [PlayerCardsArea.HandArea]: room.getPlayerById(selectedPlayers[0]).getCardIds(PlayerCardsArea.HandArea)
+            .length,
+        },
+      }),
+      ownerId,
+    );
+
+    // just for interactive
+    await room.onReceivingAsyncResponseFrom(GameEventIdentifiers.AskForChoosingCardFromPlayerEvent, ownerId);
+
+    const candidateCard = room.getPlayerById(selectedPlayers[0]).getCardIds(PlayerCardsArea.HandArea);
+    const selectedCard = candidateCard[Math.floor(Math.random() * candidateCard.length)];
+
+    room.broadcast(GameEventIdentifiers.CardDisplayEvent, {
+      displayCards: [selectedCard],
+      translationsMessage: TranslationPack.translationJsonPatcher(
+        '{0} display {1} hand card {2}',
+        TranslationPack.patchPlayerInTranslation(room.getPlayerById(ownerId)),
+        TranslationPack.patchPlayerInTranslation(room.getPlayerById(selectedPlayers[0])),
+        TranslationPack.patchCardInTranslation(selectedCard),
+      ).extract(),
+    });
+
+    room.getPlayerById(ownerId).setFlag(this.GeneralName, Sanguosha.getCardById(selectedCard).BaseType);
+
+    return true;
+  }
+}
+
+@ShadowSkill
+@CommonSkill({ name: QiangZhi.Name, description: QiangZhi.Description })
+export class QiangZhiDraw extends TriggerSkill {
+  isTriggerable(event: ServerEventFinder<GameEventIdentifiers.CardUseEvent>, stage?: AllStage) {
+    return stage === CardUseStage.CardUsing;
+  }
+
+  canUse(room: Room, owner: Player, event: ServerEventFinder<GameEventIdentifiers.CardUseEvent>) {
+    if (event.fromId !== owner.Id || room.CurrentPhasePlayer !== owner) {
+      return false;
+    }
+
+    return (
+      owner.getFlag(this.GeneralName) !== undefined &&
```

Comment:
this line is not required since if it equals the flag type, it would not be undefined